### ISSUE TITLE
Duplicate module names in Studio

### ISSUE THREAD

**Issue by SinergiaCRM** [salesagility/SuiteCRM#8747]
#### Issue
When the system has two modules with the same language label name (ex: Events), it only appears one of them in the Studio module tree (I guess it happens the same in ModuleBuilder).
This happens because it uses the "name" of the module as array key:
modules/ModuleBuilder/Module/StudioBrowser.php
```
public function getNodes()
    {
        $this->loadModules();
        $nodes = array();
        foreach ($this->modules as $module) {
            $nodes[$module->name] = $module->getNodes();
        }
        uksort($nodes, 'cmp'); // bug 15103 - order is important - this array is later looped over by foreach to generate the module list
        return $nodes;
    }
```
The property "name" isn't the internal module name but the language label:
modules/ModuleBuilder/Module/StudioModule.php
```
    public function getNodes()
    {
        return array(
            'name' => $this->name,
            'module' => $this->module,
            'type' => 'StudioModule',
            'action' => "module=ModuleBuilder&action=wizard&view_module={$this->module}",
            'children' => $this->getModule(),
            'icon' => IconRepository::getIconName($this->module)
        );
    }
```

And later it uses these nodes in:
modules/ModuleBuilder/MB/MBPackageTree.php
```
public function populateTree($nodes, &$parent)
    {
        foreach ($nodes as $node) {
            if (empty($node['label'])) {
                $node['label'] = $node['name'];
            }
            $yn = new Node($parent->id . '/' . $node['name'], $node['label']);
            if (!empty($node['action'])) {
                $yn->set_property('action', $node['action']);
            }
            $yn->set_property('href', 'javascript:void(0);');
            $yn->id = $parent->id . '/' . $node['name'];
            if (!empty($node['children'])) {
                $this->populateTree($node['children'], $yn);
            }
            $parent->add_node($yn);
        }
    }
```
It does it in this way to sort the module tree into alphabetical order.

#### Expected Behavior
Two modules with the same label name should appear in the Studio module tree.

#### Actual Behavior
It only appears one of the modules that has the same label name

#### Possible Fix
Use the internal name of the module as array key and, later, sort the modules alphabetically using the property "name" (that is the label language name of the module).

#### Steps to Reproduce
1. Change the name of the module using the functionality of the Studio inside the Labels section
2. Set a name of a module that already exists, ex: Events
3. Refresh Studio and check that only appears once the module Events.

#### Context
We are building a module that has the label name Events. But, we still want both Events module to appear on the Studio, in case we need to use either of them.

#### Your Environment
* SuiteCRM Version used: 7.11.8
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Chrome Version 51.0.2704.63 (64-bit)
* Environment name and version (e.g. MySQL, PHP 7): MariaDB PHP 7.3
* Operating System and version (e.g Ubuntu 16.04): Debian 9

**Comment by Mac-Rae** [comment]
What happens when you rename the default one? Does that make any difference?

**Comment by SinergiaCRM** [comment]
When one of the module's label name is renamed, both modules appear. But I believe they should both appear despite of the module's label name they are using.

**Comment by Mac-Rae** [comment]
Agreed, glad to hear there's a workaround. I'll look at replicating
